doc/tutorial/Server.lhs

```diff
@@ -1057,75 +1058,71 @@ into something **servant** can understand?
 If we have a function that gets us from an `m a` to an `n a`, for any `a`, what
 do we have?
 
-``` haskell ignore
-newtype m :~> n = NT { ($$) :: forall a. m a -> n a}
+``` haskell
+type (~>) m n = forall a. m a -> n a
 ```
 
 For example:
 
 ``` haskell
-listToMaybeNT :: [] :~> Maybe
-listToMaybeNT = NT listToMaybe  -- from Data.Maybe
+listToMaybe' :: [] ~> Maybe
+listToMaybe' = listToMaybe -- from Data.Maybe
 ```
 
-(`NT` comes from "natural transformation", in case you're wondering.)
+Note that `servant` doesn't declare the `~>` type-alias, as the unfolded
+variant isn't much longer to write, as we'll see shortly.
 
 So if you want to write handlers using another monad/type than `Handler`, say the `Reader String` monad, the first thing you have to
 prepare is a function:
 
 ``` haskell ignore
-readerToHandler :: Reader String :~> Handler
+readerToHandler :: Reader String a -> Handler a
 ```
 
-Let's start with `readerToHandler'`. We obviously have to run the `Reader`
-computation by supplying it with a `String`, like `"hi"`. We get an `a` out
-from that and can then just `return` it into `Handler`. We can then just wrap
-that function with the `NT` constructor to make it have the fancier type.
+We obviously have to run the `Reader` computation by supplying it with a
+`String`, like `"hi"`. We get an `a` out from that and can then just `return`
+it into `Handler`.
 
 ``` haskell
-readerToHandler' :: forall a. Reader String a -> Handler a
-readerToHandler' r = return (runReader r "hi")
-
-readerToHandler :: Reader String :~> Handler
-readerToHandler = NT readerToHandler'
+readerToHandler :: Reader String a -> Handler a
+readerToHandler r = return (runReader r "hi")
 ```
 
 We can write some simple webservice with the handlers running in `Reader String`.
 
 ``` haskell
 type ReaderAPI = "a" :> Get '[JSON] Int
-            :<|> "b" :> Get '[JSON] String
+            :<|> "b" :> ReqBody '[JSON] Double :> Get '[JSON] Bool 
 
 readerAPI :: Proxy ReaderAPI
 readerAPI = Proxy
 
 readerServerT :: ServerT ReaderAPI (Reader String)
-readerServerT = a :<|> b
-
-  where a :: Reader String Int
-        a = return 1797
+readerServerT = a :<|> b where
+    a :: Reader String Int
+    a = return 1797
 
-        b :: Reader String String
-        b = ask
+    b :: Double -> Reader String Bool
+    b _ = asks (== "hi")
 ```
 
 We unfortunately can't use `readerServerT` as an argument of `serve`, because
 `serve` wants a `Server ReaderAPI`, i.e., with handlers running in `Handler`. But there's a simple solution to this.
 
-### Enter `enter`
+### Welcome `hoistServer`
```

Comment:
Yeah, "Hoistserver `hoistServer`" and "Welcome `welcome`" just don't have that ring to them.

Reply:
I used some time, but I didn't come up with any pun on hoist :)

Reply:
Up with hoist